include/gz/utils/config.hh.in

```diff
@@ -15,3 +35,10 @@
 #cmakedefine IGNITION_UTILS_BUILD_TYPE_PROFILE 1
 #cmakedefine IGNITION_UTILS_BUILD_TYPE_DEBUG 1
 #cmakedefine IGNITION_UTILS_BUILD_TYPE_RELEASE 1
+
+namespace [[deprecated("Use gz namespace.")]]  ignition
+{
+  using namespace gz;
+}
```

Comment:
can we put this in the redirection header?

`include/ignition/utils/config.hh`?

Reply:
or we could create a new header in the `include/ignition/utils/detail/` folder with this deprecation statement and put it there, then include that detail header  in every `include/ignition` header file?

Reply:
Does the namespace definition need to have header guards in the `include/ignition/utils/config.hh`?

Reply:
I added it here without header guards. https://github.com/gazebosim/gz-utils/pull/61/commits/e1db18615bdd4771f30159b180aac72b07d95bb1

I'll do the same for the rest tomorrow.

Reply:
`deprecated_TEST` is failing, I believe it's because `config.hh` isn't included by any headers there. That's what the test was designed to catch 🎉 

```
  /github/workspace/test/integration/deprecated_TEST.cc:30:15: error: 'ignition' has not been declared
     30 |   auto impl = ignition::utils::MakeImpl<int>();
        |               ^~~~~~~~
```

Reply:
Praise Python https://github.com/gazebosim/gz-utils/pull/61/commits/960f01714fa80eab2ffd07517ca6071a604bf980